client/lib/plans/index.js

```diff
@@ -51,24 +47,38 @@ export function getPlan( planKey ) {
 	return PLANS_LIST[ planKey ];
 }
 
-export function getValidFeatureKeys() {
-	return Object.keys( FEATURES_LIST );
+export function getPlanPath( plan ) {
+	const retrievedPlan = getPlan( plan );
+	const slug = retrievedPlan.getPathSlug || ( () => undefined );
+	return slug();
```

Comment:
Nice attempt to get rid of `lodash.get` 🙂 I'm wondering if `getPathSlug` needs to be a method at all: just a simple `pathSlug` property would be sufficient. The same applies for the `getStoreSlug` and `getProductId` methods.

Reply:
Ah, this method changed files at some point during my PR, so I guess it doesn't make a lot of sense anymore to drop the `lodash.get` dependency, given that this file makes so much usage of Lodash 😁But this at least avoids string parsing, so I'll keep it 🙂 

As for `getPathSlug` and friends, I'm guessing they're kept as methods to facilitate A/B testing? I opted against changing the data structure itself, since the plans list and individual plans are being depended on directly, and it's hard to ensure there isn't any code depending on the fact that this property is a method.

Reply:
Grepping the sources for `getPathSlug`, I don't see any place where the method call would provide any extra value over a `pathSlug` property. It always returns a constant and is never overridden or customized.

It's quite common that a piece of code used to make sense some time ago but it doesn't today. Refactorings and other changes happen and not all cleanup opportunities are spotted.

Reply:
Properties are a good point. If the value needed to be determined at runtime, there would be no need for a function; a property with a getter would be enough 👍